test/eventpattern.test.ts

```diff
@@ -0,0 +1,797 @@
+import { ebEventPatternTestCase, ebEventPatternTestCaseError } from "./util";
+import { EventBusEvent, EventPredicateFunction } from "../src/eventbridge";
+import { reflect } from "../src";
+
+type TestEvent = EventBusEvent<{
+  num: number;
+  str: string;
+  optional?: string;
+  "multi-part": string;
+  deep: {
+    value: string;
+  };
+  bool: boolean;
+  array: string[];
+  numArray: number[];
+  boolArray: boolean[];
+}>;
+
+describe("event pattern", () => {
+  describe("equals", () => {
+    test("simple", () => {
+      ebEventPatternTestCase(
+        reflect<EventPredicateFunction<TestEvent>>(
+          (event) => event.source === "lambda"
+        ),
+        {
+          source: ["lambda"],
+        }
+      );
+    });
+
+    test("double equals", () => {
+      ebEventPatternTestCase(
+        reflect<EventPredicateFunction<TestEvent>>(
+          (event) => event.source == "lambda"
```

Comment:
Can we provide a union type of string literals or an enum to provide more guidance here on the allowed values while also leaving it open for all types of integrations?

Reply:
You mean for source?

Reply:
yes for `event.source` - is that a closed set of values?

Reply:
No, it can be any string value, and some AWS services send their ID. I'd say we can add a list later, but we'll always allow for a set of strings.